### ISSUE TITLE
renderEvents not working?

### ISSUE THREAD

**Issue by jonasgroendahl** [fullcalendar/fullcalendar#3477]
Hi,

Am I the only one who can't get renderEvents to work? In 3.1 they added this to improve performance since calling renderEvent in a for loop for hundreds of events makes the site freeze while it adds events. $("#calendar").fullCalendar("addEventSource", events)  seems like the best alternative atm to improve performance since renderEvents isn't working for me. The function is being called but the events aren't added onto the calendar. renderEvent works just fine but as I said this is bad performance wise.

**Comment by arshaw** [comment]
Would you be able to post a [runnable, stripped-down demonstration of the bug](http://fullcalendar.io/wiki/Reduced-Test-Cases/)? Would really appreciate it because the time saved reproducing will be time spent fixing.

**Comment by Chris136** [comment]
Maybe I have an idea why @jonasgroendahl cannot use renderEvent properly.
I had this behavior in my code, too. My code was working in version 3.0.0 and it did not work in 3.6.1

In the events function I called the callback only if > 0 events in my array.
I noticed that fullCalendar had set the internal variable "pendingCnt" to "1" and skipped rendering because it still was waiting for the events fetch as the callback never was executed.

Long story short:
Executing the events callback with an empty array fixed my renderEvent problem!
